### ISSUE TITLE
LS Startup Randomly fails on java.lang.BootstrapMethodError (Reliable Reproducer Inside now)

### ISSUE THREAD

**Issue by original-brownbear** [elastic/logstash#7999]
Update: Use 8 threads `-w 8` and it fails close to 100% of the time!

One reliable failure I can see around JRuby JIT is this one:

```sh
cat ~/Downloads/apache_access_logs ~/Downloads/apache_access_logs | bin/logstash -b 128 -w 4 -f ~/tmp/logstash.cfg 
Unhandled Java exception: java.lang.BootstrapMethodError: java.lang.NoSuchMethodError: Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$block$output_batch$3(Lorg/jruby/runtime/ThreadContext;Lorg/jruby/runtime/Block;Lorg/jruby/parser/StaticScope;Lorg/jruby/runtime/builtin/IRubyObject;[Lorg/jruby/runtime/builtin/IRubyObject;Lorg/jruby/runtime/Block;Ljava/lang/String;Lorg/jruby/runtime/Block$Type;)Lorg/jruby/runtime/builtin/IRubyObject;
java.lang.BootstrapMethodError: java.lang.NoSuchMethodError: Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$block$output_batch$3(Lorg/jruby/runtime/ThreadContext;Lorg/jruby/runtime/Block;Lorg/jruby/parser/StaticScope;Lorg/jruby/runtime/builtin/IRubyObject;[Lorg/jruby/runtime/builtin/IRubyObject;Lorg/jruby/runtime/Block;Ljava/lang/String;Lorg/jruby/runtime/Block$Type;)Lorg/jruby/runtime/builtin/IRubyObject;
                  block in output_batch at /Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:529
                             callDirect at org/jruby/runtime/CompiledIRBlockBody.java:145
                             callDirect at org/jruby/runtime/MixedModeIRBlockBody.java:113
                                   call at org/jruby/runtime/IRBlockBody.java:71
                                   call at org/jruby/runtime/Block.java:124
                                   call at org/jruby/RubyProc.java:289
                                 call19 at org/jruby/RubyProc.java:273
                          block in each at /Users/brownbear/src/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:227
                            yieldDirect at org/jruby/runtime/CompiledIRBlockBody.java:156
                                  yield at org/jruby/runtime/BlockBody.java:114
                                  yield at org/jruby/runtime/Block.java:165
                                   each at org/jruby/javasupport/ext/JavaLang.java:93
                                   each at org/jruby/javasupport/ext/JavaUtil.java:133
                                   each at /Users/brownbear/src/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:226
         RUBY$method$each$0$__VARARGS__ at Users/brownbear/src/logstash/logstash_minus_core/lib/logstash/util//Users/brownbear/src/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:-1
                                   call at org/jruby/internal/runtime/methods/CompiledIRMethod.java:77
                                   call at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:93
                              callBlock at org/jruby/runtime/callsite/CachingCallSite.java:77
                                   call at org/jruby/runtime/callsite/CachingCallSite.java:83
                              interpret at org/jruby/ir/instructions/CallBase.java:428
                            processCall at org/jruby/ir/interpreter/InterpreterEngine.java:355
                              interpret at org/jruby/ir/interpreter/StartupInterpreterEngine.java:73
                       INTERPRET_METHOD at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:109
                                   call at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:95
                                 invoke at org/jruby/ir/targets/InvokeSite.java:145
                            worker_loop at /Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:481
  RUBY$method$worker_loop$0$__VARARGS__ at Users/brownbear/src/logstash/logstash_minus_core/lib/logstash//Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:-1
                                   call at org/jruby/internal/runtime/methods/CompiledIRMethod.java:77
                                   call at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:93
                                 invoke at org/jruby/ir/targets/InvokeSite.java:145
                 block in start_workers at /Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:437
                             callDirect at org/jruby/runtime/CompiledIRBlockBody.java:145
                                   call at org/jruby/runtime/IRBlockBody.java:71
                                   call at org/jruby/runtime/Block.java:124
                                   call at org/jruby/RubyProc.java:289
                                   call at org/jruby/RubyProc.java:246
                                    run at org/jruby/internal/runtime/RubyRunnable.java:104
                                    run at java/lang/Thread.java:748

Caused by:
java.lang.NoSuchMethodError: Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$block$output_batch$3(Lorg/jruby/runtime/ThreadContext;Lorg/jruby/runtime/Block;Lorg/jruby/parser/StaticScope;Lorg/jruby/runtime/builtin/IRubyObject;[Lorg/jruby/runtime/builtin/IRubyObject;Lorg/jruby/runtime/Block;Ljava/lang/String;Lorg/jruby/runtime/Block$Type;)Lorg/jruby/runtime/builtin/IRubyObject;
                                resolve at java/lang/invoke/MethodHandleNatives.java:-2
                                resolve at java/lang/invoke/MemberName.java:975
                          resolveOrFail at java/lang/invoke/MemberName.java:1000
                          resolveOrFail at java/lang/invoke/MethodHandles.java:1394
               linkMethodHandleConstant at java/lang/invoke/MethodHandles.java:1750
               linkMethodHandleConstant at java/lang/invoke/MethodHandleNatives.java:477
                  block in output_batch at /Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:529
                             callDirect at org/jruby/runtime/CompiledIRBlockBody.java:145
                             callDirect at org/jruby/runtime/MixedModeIRBlockBody.java:113
                                   call at org/jruby/runtime/IRBlockBody.java:71
                                   call at org/jruby/runtime/Block.java:124
                                   call at org/jruby/RubyProc.java:289
                                 call19 at org/jruby/RubyProc.java:273
                          block in each at /Users/brownbear/src/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:227
                            yieldDirect at org/jruby/runtime/CompiledIRBlockBody.java:156
                                  yield at org/jruby/runtime/BlockBody.java:114
                                  yield at org/jruby/runtime/Block.java:165
                                   each at org/jruby/javasupport/ext/JavaLang.java:93
                                   each at org/jruby/javasupport/ext/JavaUtil.java:133
                                   each at /Users/brownbear/src/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:226
         RUBY$method$each$0$__VARARGS__ at Users/brownbear/src/logstash/logstash_minus_core/lib/logstash/util//Users/brownbear/src/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:-1
                                   call at org/jruby/internal/runtime/methods/CompiledIRMethod.java:77
                                   call at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:93
                              callBlock at org/jruby/runtime/callsite/CachingCallSite.java:77
                                   call at org/jruby/runtime/callsite/CachingCallSite.java:83
                              interpret at org/jruby/ir/instructions/CallBase.java:428
                            processCall at org/jruby/ir/interpreter/InterpreterEngine.java:355
                              interpret at org/jruby/ir/interpreter/StartupInterpreterEngine.java:73
                       INTERPRET_METHOD at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:109
                                   call at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:95
                                 invoke at org/jruby/ir/targets/InvokeSite.java:145
                            worker_loop at /Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:481
  RUBY$method$worker_loop$0$__VARARGS__ at Users/brownbear/src/logstash/logstash_minus_core/lib/logstash//Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:-1
                                   call at org/jruby/internal/runtime/methods/CompiledIRMethod.java:77
                                   call at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:93
                                 invoke at org/jruby/ir/targets/InvokeSite.java:145
                 block in start_workers at /Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:437
                             callDirect at org/jruby/runtime/CompiledIRBlockBody.java:145
                                   call at org/jruby/runtime/IRBlockBody.java:71
                                   call at org/jruby/runtime/Block.java:124
                                   call at org/jruby/RubyProc.java:289
                                   call at org/jruby/RubyProc.java:246
                                    run at org/jruby/internal/runtime/RubyRunnable.java:104
                                    run at java/lang/Thread.java:748

```

with config 

```
input {
  stdin { }
}

filter {
  grok {
    match => {
      "message" => '%{IPORHOST:clientip} %{USER:ident} %{USER:auth} \[%{HTTPDATE:timestamp}\] "%{WORD:verb} %{DATA:request} HTTP/%{NUMBER:httpversion}" %{NUMBER:response:int} (?:-|%{NUMBER:bytes:int}) %{QS:referrer} %{QS:agent}'
    }
  }

  date {
    match => [ "timestamp", "dd/MMM/YYYY:HH:mm:ss Z" ]
    locale => fr
  }

  geoip {
    source => "clientip"
  }

  useragent {
    source => "agent"
    target => "useragent"
  }
}

output {
  stdout { codec => "dots"}
}
```

This happens for about 25% of runs in `master`! Fixed by #7950 though :P ... but this seems really bad for `6.0` :(

**Comment by original-brownbear** [comment]
one some runs it also goes into an endless loop of this error being printed:

```
[2017-08-15T23:56:47,732][WARN ][logstash.filters.grok    ] Grok regexp threw exception {:exception=>"org.jruby.runtime.scopes.DynamicScope1 only supports scopes with 1 variables", :backtrace=>["org.jruby.runtime.scopes.DynamicScope1.sizeError(Unknown Source)", "org.jruby.runtime.scopes.DynamicScope1.getValueDepthZero(Unknown Source)", "org.jruby.runtime.DynamicScope.getValueDepthZeroOrNil(DynamicScope.java:257)", "org.jruby.runtime.DynamicScope.getValueOrNil(DynamicScope.java:249)", "org.jruby.runtime.DynamicScope.getValueOrNil(DynamicScope.java:247)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$block$match_against_groks$1(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:348)", "org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:156)", "org.jruby.runtime.IRBlockBody.yieldSpecific(IRBlockBody.java:80)", "org.jruby.runtime.Block.yieldSpecific(Block.java:134)", "org.jruby.ir.runtime.IRRuntimeHelpers.yieldSpecific(IRRuntimeHelpers.java:419)", "org.jruby.ir.targets.YieldSite.yieldSpecific(YieldSite.java:114)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.timeout_enforcer.RUBY$method$grok_till_timeout$0(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok/timeout_enforcer.rb:21)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$block$match_against_groks$1(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:348)", "org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:156)", "org.jruby.runtime.BlockBody.yield(BlockBody.java:114)", "org.jruby.runtime.Block.yield(Block.java:165)", "org.jruby.RubyArray.each(RubyArray.java:1734)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$method$match_against_groks$0(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:345)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$method$match$0(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:334)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$block$filter$1(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:300)", "org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:156)", "org.jruby.runtime.BlockBody.yield(BlockBody.java:114)", "org.jruby.runtime.Block.yield(Block.java:165)", "org.jruby.RubyHash$12.visit(RubyHash.java:1362)", "org.jruby.RubyHash$12.visit(RubyHash.java:1359)", "org.jruby.RubyHash.visitLimited(RubyHash.java:662)", "org.jruby.RubyHash.visitAll(RubyHash.java:647)", "org.jruby.RubyHash.iteratorVisitAll(RubyHash.java:1319)", "org.jruby.RubyHash.each_pairCommon(RubyHash.java:1354)", "org.jruby.RubyHash.each(RubyHash.java:1343)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$method$filter$0(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:299)", "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.filters.base.RUBY$method$do_filter$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/filters/base.rb:145)", "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.filters.base.RUBY$block$multi_filter$1(/Users/brownbear/src/logstash/logstash-core/lib/logstash/filters/base.rb:164)", "org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:156)", "org.jruby.runtime.BlockBody.yield(BlockBody.java:114)", "org.jruby.runtime.Block.yield(Block.java:165)", "org.jruby.RubyArray.each(RubyArray.java:1734)", "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.filters.base.RUBY$method$multi_filter$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/filters/base.rb:161)", "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.filter_delegator.RUBY$method$multi_filter$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/filter_delegator.rb:48)", "org.jruby.internal.runtime.methods.CompiledIRMethod.call(CompiledIRMethod.java:103)", "org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:163)", "org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:200)", "org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:161)", "org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:314)", "org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)", "org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:132)", "org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:148)", "org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:73)", "org.jruby.runtime.Block.call(Block.java:124)", "org.jruby.RubyProc.call(RubyProc.java:289)", "org.jruby.internal.runtime.methods.ProcMethod.call(ProcMethod.java:63)", "org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:204)", "org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:200)", "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$block$filter_batch$1(/Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:499)", "org.jruby.runtime.CompiledIRBlockBody.callDirect(CompiledIRBlockBody.java:145)", "org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:71)", "org.jruby.runtime.Block.call(Block.java:124)", "org.jruby.RubyProc.call(RubyProc.java:289)", "org.jruby.RubyProc.call19(RubyProc.java:273)", "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.util.wrapped_synchronous_queue.RUBY$block$each$1(/Users/brownbear/src/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:227)", "org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:156)", "org.jruby.runtime.BlockBody.yield(BlockBody.java:114)", "org.jruby.runtime.Block.yield(Block.java:165)", "org.jruby.javasupport.ext.JavaLang$Iterable.each(JavaLang.java:93)", "org.jruby.javasupport.ext.JavaUtil$Collection.each(JavaUtil.java:133)", "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.util.wrapped_synchronous_queue.RUBY$method$each$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:226)", "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$method$filter_batch$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:498)", "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$method$worker_loop$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:477)", "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$block$start_workers$2(/Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:437)", "org.jruby.runtime.CompiledIRBlockBody.callDirect(CompiledIRBlockBody.java:145)", "org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:71)", "org.jruby.runtime.Block.call(Block.java:124)", "org.jruby.RubyProc.call(RubyProc.java:289)", "org.jruby.RubyProc.call(RubyProc.java:246)", "org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:104)", "java.lang.Thread.run(Thread.java:748)"], :class=>"Java::JavaLang::RuntimeException"}
```

**Comment by original-brownbear** [comment]
Attached 2 jit logs, one for the failed one for the sucessful startup case
[failed.jit.txt](https://github.com/elastic/logstash/files/1227224/failed.jit.txt)
[sucessful.jit.txt](https://github.com/elastic/logstash/files/1227223/sucessful.jit.txt)

**Comment by original-brownbear** [comment]
```rb
{ :exception => "org.jruby.runtime.scopes.DynamicScope1 only supports scopes with 1 variables",
  :backtrace => [
    "org.jruby.runtime.scopes.DynamicScope1.sizeError(Unknown Source)",
    "org.jruby.runtime.scopes.DynamicScope1.getValueDepthZero(Unknown Source)",
    "org.jruby.runtime.DynamicScope.getValueDepthZeroOrNil(DynamicScope.java:257)",
    "org.jruby.runtime.DynamicScope.getValueOrNil(DynamicScope.java:249)",
    "org.jruby.runtime.DynamicScope.getValueOrNil(DynamicScope.java:247)",
    "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$block$match_against_groks$1(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:348)", "org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:156)", "org.jruby.runtime.IRBlockBody.yieldSpecific(IRBlockBody.java:80)", "org.jruby.runtime.Block.yieldSpecific(Block.java:134)", "org.jruby.ir.runtime.IRRuntimeHelpers.yieldSpecific(IRRuntimeHelpers.java:419)", "org.jruby.ir.targets.YieldSite.yieldSpecific(YieldSite.java:114)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.timeout_enforcer.RUBY$method$grok_till_timeout$0(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok/timeout_enforcer.rb:21)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$block$match_against_groks$1(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:348)", "org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:156)", "org.jruby.runtime.BlockBody.yield(BlockBody.java:114)", "org.jruby.runtime.Block.yield(Block.java:165)", "org.jruby.RubyArray.each(RubyArray.java:1734)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$method$match_against_groks$0(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:345)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$method$match$0(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:334)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$block$filter$1(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:300)", "org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:156)", "org.jruby.runtime.BlockBody.yield(BlockBody.java:114)", "org.jruby.runtime.Block.yield(Block.java:165)", "org.jruby.RubyHash$12.visit(RubyHash.java:1362)", "org.jruby.RubyHash$12.visit(RubyHash.java:1359)", "org.jruby.RubyHash.visitLimited(RubyHash.java:662)", "org.jruby.RubyHash.visitAll(RubyHash.java:647)", "org.jruby.RubyHash.iteratorVisitAll(RubyHash.java:1319)", "org.jruby.RubyHash.each_pairCommon(RubyHash.java:1354)", "org.jruby.RubyHash.each(RubyHash.java:1343)", "Users.brownbear.src.logstash.vendor.bundle.jruby.$2_dot_3_dot_0.gems.logstash_minus_filter_minus_grok_minus_3_dot_4_dot_2.lib.logstash.filters.grok.RUBY$method$filter$0(/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-grok-3.4.2/lib/logstash/filters/grok.rb:299)",
    "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.filters.base.RUBY$method$do_filter$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/filters/base.rb:145)",
    "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.filters.base.RUBY$block$multi_filter$1(/Users/brownbear/src/logstash/logstash-core/lib/logstash/filters/base.rb:164)",
    "org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:156)",
    "org.jruby.runtime.BlockBody.yield(BlockBody.java:114)",
    "org.jruby.runtime.Block.yield(Block.java:165)",
    "org.jruby.RubyArray.each(RubyArray.java:1734)",
    "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.filters.base.RUBY$method$multi_filter$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/filters/base.rb:161)",
    "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.filter_delegator.RUBY$method$multi_filter$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/filter_delegator.rb:48)",
    "org.jruby.internal.runtime.methods.CompiledIRMethod.call(CompiledIRMethod.java:103)",
    "org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:163)",
    "org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:200)",
    "org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:161)",
    "org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:314)",
    "org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)",
    "org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:132)",
    "org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:148)",
    "org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:73)",
    "org.jruby.runtime.Block.call(Block.java:124)",
    "org.jruby.RubyProc.call(RubyProc.java:289)",
    "org.jruby.internal.runtime.methods.ProcMethod.call(ProcMethod.java:63)",
    "org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:204)",
    "org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:200)",
    "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$block$filter_batch$1(/Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:499)",
    "org.jruby.runtime.CompiledIRBlockBody.callDirect(CompiledIRBlockBody.java:145)",
    "org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:71)", "org.jruby.runtime.Block.call(Block.java:124)",
    "org.jruby.RubyProc.call(RubyProc.java:289)", "org.jruby.RubyProc.call19(RubyProc.java:273)",
    "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.util.wrapped_synchronous_queue.RUBY$block$each$1(/Users/brownbear/src/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:227)",
    "org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:156)",
    "org.jruby.runtime.BlockBody.yield(BlockBody.java:114)",
    "org.jruby.runtime.Block.yield(Block.java:165)",
    "org.jruby.javasupport.ext.JavaLang$Iterable.each(JavaLang.java:93)",
    "org.jruby.javasupport.ext.JavaUtil$Collection.each(JavaUtil.java:133)",
    "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.util.wrapped_synchronous_queue.RUBY$method$each$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:226)",
    "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$method$filter_batch$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:498)",
    "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$method$worker_loop$0(/Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:477)",
    "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$method$worker_loop$0$__VARARGS__(/Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb)",
    "org.jruby.internal.runtime.methods.CompiledIRMethod.call(CompiledIRMethod.java:77)",
    "org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:93)",
    "org.jruby.ir.targets.InvokeSite.invoke(InvokeSite.java:145)",
    "Users.brownbear.src.logstash.logstash_minus_core.lib.logstash.pipeline.RUBY$block$start_workers$2(/Users/brownbear/src/logstash/logstash-core/lib/logstash/pipeline.rb:437)",
    "org.jruby.runtime.CompiledIRBlockBody.callDirect(CompiledIRBlockBody.java:145)",
    "org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:71)",
    "org.jruby.runtime.Block.call(Block.java:124)",
    "org.jruby.RubyProc.call(RubyProc.java:289)",
    "org.jruby.RubyProc.call(RubyProc.java:246)",
    "org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:104)",
    "java.lang.Thread.run(Thread.java:748)"
  ], :class => "Java::JavaLang::RuntimeException"
}
```

Clean failure stacktrace

**Comment by original-brownbear** [comment]
Created https://github.com/jruby/jruby/issues/4739

**Comment by original-brownbear** [comment]
None Logstash reproducer 

https://github.com/original-brownbear/jruby-jit-issue-reproducer#jit-dynamicscope-size-error (code is in that repo, link points at the error reproduced the the command used to trigger it )

**Comment by headius** [comment]
This may be fixed by the fixes in #4739.

**Comment by original-brownbear** [comment]
@headius thanks so much for looking into this so quickly, will test fixes very soon.

**Comment by original-brownbear** [comment]
@headius sorry for the delay got around to testing this now and it's looking good as far as I can see! I'd say the chance this fix at least significantly ups stability is `>95%` so I'd love to see a `9.1.13.0` with this included :)

**Comment by headius** [comment]
jruby/jruby#4739 will be in 9.1.13, which should see a release this week or next.

**Comment by original-brownbear** [comment]
fixed thanks to JRuby `9.1.13.0` :)
